### PR TITLE
Add loading of collections with faulty components

### PR DESCRIPTION
**Pull request by sophie-h**

### CODE REVIEW

File: src/collection.rs [hunk 1]
```diff
@@ -35,8 +43,26 @@ impl Collection {
...
+
+    /// Create a new `Collection` from an XML file; skipping invalid components.
+    ///
+    /// For each invalid component, an error is collected into the returned
+    /// value.
+    ///
+    /// # Arguments
+    ///
+    /// * `path` - The path to the collection.
+    pub fn from_path_with_partial(
```
Comment (bilelmoussaoui) [inline src/collection.rs:57]: Can we get a similar method for the gzipped variant please?

File: src/error.rs [hunk 2]
```diff
@@ -75,3 +79,106 @@ impl ParseError {
...
+    let mut code = Vec::new();
+    let _ = context.write_with_config(
+        &mut code,
+        xmltree::EmitterConfig::new()
+            .write_document_declaration(false)
+            .perform_indent(true),
+    );
+    let code_string = String::from_utf8_lossy(&code).to_string();
+
+    match code_string.char_indices().nth(200) {
```
Comment (bilelmoussaoui) [inline src/error.rs:180]: Why the 200th exactly? i would write a single line comment about the reasoning if there is any

File: src/collection.rs (hunk 1 above)
Comment (sophie-h) [reply src/collection.rs:57]: Skipping this since I think they don't work anyways

File: src/error.rs (hunk 2 above)
Comment (sophie-h) [reply src/error.rs]: Added a comment

**Pull request merged**